loom/script/native/core/system/lmDate.cpp

```diff
@@ -18,33 +18,102 @@
 * ===========================================================================
 */
 
+#include "loom/script/loomscript.h"
 #include "loom/script/native/lsLuaBridge.h"
 #include <ctime>
 
 using namespace LS;
 
+// Define a macro for getters and setters. There are a bunch of them!
+#define DATE_GETTER_SETTER(var, tm_var) \
+int get ## var() { return timeInfo.tm_ ## tm_var; } \
+void set ## var(int value) { timeInfo.tm_ ## tm_var = value; } \
+int get ## var ## UTC() { return timeInfoUTC.tm_ ## tm_var; } \
+void set ## var ## UTC(int value) { timeInfoUTC.tm_ ## tm_var = value; }
+
 class Date {
 private:
     // The buffer size when returning string data
     static const int BUFFER_SIZE = 1024;
 
+    // Private variables, their fields will be accessed by public methods
+    struct tm timeInfo;    // The time info for this Date instance. It will be accessed by various getters and setters in ls
+    struct tm timeInfoUTC; // Same as timeInfo, but for UTC instead of local time
+
 public:
-    static const char *formatTime(const char *format)
+    // Static: Determine if the provided year is a leap year
+    static bool isLeapYear(int year)
     {
-        time_t rawtime;
-        struct tm *timeinfo;
-        char buffer[BUFFER_SIZE];
+        if (year % 4 != 0)
+            return false;
+        else if (year % 100 == 0 && year % 400 != 0)
+            return false;
+        else
+            return true;
+    }
 
+    Date()
+    {
+        time_t rawtime;
+        
         time(&rawtime);
-        timeinfo = localtime(&rawtime);
 
-        strftime(buffer, BUFFER_SIZE - 1, format, timeinfo);
+        // Initialize tm structs
+        timeInfo = *localtime(&rawtime);
+        timeInfoUTC = *gmtime(&rawtime);
+    }
+
+    // Getters and setters for the local timeInfo
+    DATE_GETTER_SETTER(Date, mday);
+    DATE_GETTER_SETTER(Day, wday);
+    DATE_GETTER_SETTER(Hours, hour);
+    DATE_GETTER_SETTER(Minutes, min);
+    DATE_GETTER_SETTER(Month, mon);
+    DATE_GETTER_SETTER(Seconds, sec);
+
+    // Year getters and setters must be defined seperately because of an offset
+    int getYear() 
+    { 
+        return timeInfo.tm_year + 1900;
+    }
+    void setYear(int value) 
+    { 
+        timeInfo.tm_year = value - 1900;
+    }
+
+    int getYearUTC() 
+    { 
+        return timeInfoUTC.tm_year + 1900;
+    }
+    void setYearUTC(int value) 
+    { 
+        timeInfoUTC.tm_year = value - 1900;
+    }
+
+
+    const char *formatTime(const char *format)
+    {
+        char buffer[BUFFER_SIZE];
+
+        strftime(buffer, BUFFER_SIZE - 1, format, &timeInfo);
         
         // The buffer must be converted into a const char* before returning
         const char *retString = buffer;
 
         return retString;
```

Comment:
You can do this cast in one line, return (const char *)buffer;, which clarifies that it's an inconsequential cast.